src/lib/update-cors-cloud.ts

```diff
@@ -0,0 +1,112 @@
+import axios, { AxiosResponse } from 'axios';
+import { ADD_CONFIG, GET_CONFIG } from './constants';
+
+function handleOrigins(originsToAdd: string[], origins: string[], remove: boolean) {
```

Comment:
I'm struggling a bit to understand what this function is intended to do. The function name is `handleOrigins` (what does "handle" mean?), we have `originsToAdd` (okay, so this is used to add additional origins to an existing list of origins?) and then we have `origins` and then a `remove` option (huh?). You then have the comment below, but I'm afraid it doesn't clear things up for me.

Reply:
There are two parts to this process.

Part 1: Adding a list of new urls to an existing cors configuration

Part 2: Removing a list of urls from a cors configuration. Setting the value `remove` in the action instructs the action to remove the urls. This is basically just doing a PUT to the cors configuration and not including the urls in that list on `acceptedOrigins` so this here says, ok, remove the origins from the existing cors configuration.

To do that we make a request for the cors configuration and get the current urls from AM, then we `filter` that and say does this url match? cool remove it. (`rejects`) basically.